### PR TITLE
Fixed DNS messed up bug

### PR DESCRIPTION
**Pull request by lianke123321**

### CODE REVIEW

**Comment by lianke123321** [comment]
@rpanah can you review and merge? Also I think it's a good idea to release a small new version because of this

**Comment by rpanah** [comment]
@lianke123321 did you test it using the method I suggested?

**Comment by lianke123321** [comment]
I did. Using top-500 list.

File: centinel/primitives/dnslib.py [hunk 1]
```diff
@@ -158,7 +158,6 @@ def lookup_domain(self, domain, nameserver=None, log_prefix=''):
         results = {'domain': domain, 'nameserver': nameserver}
         # construct the socket to use
         sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
```
Comment (rpanah) [inline centinel/primitives/dnslib.py]: What is `socket.SOL_SOCKET`, and is removing it safe?

File: centinel/primitives/dnslib.py [hunk 2]
```diff
@@ -171,36 +170,58 @@ def lookup_domain(self, domain, nameserver=None, log_prefix=''):
...
+            resp = dns.message.from_wire(response)
+            results['response1-ips'] = parse_out_ips(resp)
+
+            # first domain name in response should be the same with query
+            # domain name
+            for entry in resp.answer:
+                if domain.lower() != entry.name.to_text().lower()[:-1]:
+                    logging.debug("%sWrong domain name %s for %s!"
+                                  % (log_prefix, entry.name.to_text().lower()[:-1], domain))
+                    results['response1-domain'] = entry.name.to_text().lower()[:-1]
```
Comment (rpanah) [inline centinel/primitives/dnslib.py]: @lianke123321 are you storing the actual result with the wrong query domain name somewhere?

Comment (lianke123321) [inline centinel/primitives/dnslib.py]: I am, at this line.

File: centinel/primitives/dnslib.py (hunk 1 above)
Comment (lianke123321) [inline centinel/primitives/dnslib.py]: socket.SOL_SOCKET indicates the level of the parameter being set. In this case, socket.SO_REUSEADDR is a socket.SOL_SOCKET level parameter. This whole line means setting socket.SO_REUSEADDR to 1.

**Pull request merged**